FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+import { Link } from 'react-router-dom';
+import { StarOutlined, StarFilled } from '@ant-design/icons';
+import { useState, useMemo } from 'react';
+import { useSWRConfig } from 'swr';
+import useSWRMutation from 'swr/mutation';
+import styles from './CompanyCard.module.css';
+import { useUser } from '../../hooks';
+import PropTypes from 'prop-types';
+import { Tooltip, Badge } from 'antd';
+
+export default function CompanyCard({ data, isAuthorized }) {
+  CompanyCard.propTypes = {
+    data: PropTypes.object,
+    isAuthorized: PropTypes.bool,
+  };
+  const [isSaved, setIsSaved] = useState(data && data.is_saved);
+  const { mutate } = useSWRConfig();
+  const profile = useMemo(() => {
+    return {
+      id: data?.id,
+      personId: data?.person,
+      name: data?.name,
+      region: data?.region,
+      categories: data?.categories,
+      activities: data?.activities,
+      logo: data?.logo_image,
+      banner: data?.banner_image,
+      founded: data?.founded,
+      isSaved: data?.is_saved,
+    };
+  }, [data]);
+
+  const { user } = useUser();
+  const ownProfile = user && user.id === profile.personId;
+
+  async function sendRequest(url, { arg: data }) {
+    const authToken = localStorage.getItem('Token');
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Token ${authToken}`,
+      },
+      body: JSON.stringify(data),
+    }).then();
+  }
+
+  const { trigger } = useSWRMutation(
+    `${process.env.REACT_APP_BASE_API_URL}/api/saved-list/`,
+    sendRequest
+  );
+
+  const handleClick = async () => {
+    try {
+      await trigger(
+        { company_pk: profile.id },
+        { optimisticData: () => setIsSaved(!isSaved) }
+      );
+      mutate(
+        (key) => typeof key === 'string' && key.startsWith('/api/profiles/'),
```

Comment:
Should we the exact api url to update it? I think `/api/profiles/` might cause mutation for a few keys.

Reply:
Thank your for feedback. Done.